src/content/docs/zh-cn/core-concepts/astro-syntax.mdx

```diff
@@ -0,0 +1,191 @@
+---
+title: Astro 语法
+description: .astro 组件语法简介。
+i18nReady: true
+---
+
+**如果你了解 HTML，那么你已经足够了解编写第一个 Astro 组件。**
+
+Astro 组件语法是 HTML 的超集。这个语法是[设计用来让任何有编写 HTML 或 JSX 经验的人感到熟悉](#astro-和-jsx-间的差异) ，并增加了对包含组件和 JavaScript 表达式的支持。
+
+## 类 JSX 表达式
+
+你可以在 Astro 组件的两个代码围栏(`---`)之间的 frontmatter 组件脚本内部定义本地 JavaScript 变量。然后，可以使用类似 JSX 的表达式将这些变量注入到组件的 HTML 模板中！
+
+:::note[动态 vs 响应式]
+
+使用这种方法，你可以在 frontmatter 里设置计算的**动态**值。但是一旦设置完，这些值就不是**响应式**的了，也永远不会改变。Astro 组件是在渲染阶段中只运行一次的模板。
+
+请参阅下面的更多例子[Astro 和 JSX 之间的差异](#astro-和-jsx-间的差异)。
+:::
+
+### 变量
+
+可以使用花括号语法将局部变量添加到 HTML 中:
+
+```astro title="src/components/Variables.astro" "{name}"
+---
+const name = "Astro";
+---
+<div>
+  <h1>Hello {name}!</h1>  <!-- 输出 <h1>Hello Astro!</h1> -->
+</div>
+```
+
+### 动态属性
+
+可以在花括号中使用局部变量将属性值传递给 HTML 元素和组件:
+
+```astro title="src/components/DynamicAttributes.astro" "{name}" "${name}"
+---
+const name = "Astro";
+---
+<h1 class={name}>Attribute expressions are supported</h1>
+
+<MyComponent templateLiteralNameAttribute={`MyNameIs${name}`} />
+```
+
+:::caution
+HTML 属性将转换为字符串，因此不可能将函数和对象传递给 HTML 元素。
+例如，不能为 Astro 组件中的 HTML 元素分配事件处理程序:
+
+```astro
+---
+// dont-do-this.astro
+function handleClick () {
+    console.log("button clicked!");
+}
+---
+<!-- ❌ 这样行不通 ❌ -->
+<button onClick={handleClick}>Nothing will happen when you click me!</button>
+```
+
+相反，使用客户端脚本来添加事件处理程序，就像在普通的 JavaScript 中那样:
+
+```astro
+---
+// do-this-instead.astro
+---
+<button id="button">Click Me</button>
+<script>
+  function handleClick () {
+    console.log("button clicked!");
+  }
+  document.getElementById("button").addEventListener("click", handleClick);
+</script>
+```
+:::
+
+### 动态 HTML
+
+可以在类 JSX 函数中使用局部变量来生成动态生成的 HTML 元素:
+
+```astro title="src/components/DynamicHtml.astro" "{item}"
+---
+const items = ["Dog", "Cat", "Platypus"];
+---
+<ul>
+  {items.map((item) => (
+    <li>{item}</li>
+  ))}
+</ul>
+```
+
+Astro 可以使用 JSX 逻辑运算符和三元表达式有条件地显示 HTML。
+
+```astro title="src/components/ConditionalHtml.astro" "visible"
+---
+const visible = true;
+---
+{visible && <p>Show me!</p>}
+
+{visible ? <p>Show me!</p> : <p>Else show me!</p>}
+```
+
+### 动态标签
+
+还可以通过将变量设置为 HTML 标签或组件导入来使用动态标签:
+
+```astro title="src/components/DynamicTags.astro" /Element|(?<!My)Component/
+---
+import MyComponent from "./MyComponent.astro";
+const Element = 'div'
+const Component = MyComponent;
+---
+<Element>Hello!</Element> <!-- 渲染成 <div>Hello!</div> -->
+<Component /> <!-- 渲染成 <MyComponent /> -->
+```
+
+当使用动态标签时：
+
+- **变量名必须大**。例如，使用 `Element`，而不是 `element`。否则，Astro 将尝试把变量名渲染为文本 HTML 标签。
+
+- **不支持 hydration 指令**。当使用[`client:*` hydration 指令时](/zh-cn/core-concepts/framework-components/#激活组件)，Astro 需要知道哪些组件要捆绑到生产环境中，而动态标签模式阻止了这一点。
+
+### 片段及多元素
```

Comment:
Some content is outdated. See commit: https://github.com/withastro/docs/commit/6b6f629e2241dce12dd4763124d00eeb223aa1f4